frame/staking/src/lib.rs

```diff
@@ -2052,11 +2069,13 @@ impl<T: Config> Module<T> {
 			let nominator_reward: BalanceOf<T> = nominator_exposure_part * validator_leftover_payout;
 			// We can now make nominator payout:
 			if let Some(imbalance) = Self::make_payout(&nominator.who, nominator_reward) {
+				// Note: this logic does not count payouts for `RewardDestination::None`.
+				payout_count += 1;
 				Self::deposit_event(RawEvent::Reward(nominator.who.clone(), imbalance.peek()));
 			}
 		}
 
-		Ok(())
+		Ok(Some(T::WeightInfo::payout_stakers_alive_staked(payout_count)).into())
```

Comment:
```suggestion
		debug_assert!(payout_count <= MaxNominatorRewardedPerValidator::get());
		Ok(Some(T::WeightInfo::payout_stakers_alive_staked(payout_count)).into())
```

no particular reason :D

Reply:
+1 for validator

Reply:
I added this - just to confirm my understanding of `debug_assert`
1) they don't make it into wasm?
1) they don't make it into `--release` builds?

Reply:
They don't make it into release. They could still be in was made from debug